sklearn/manifold/_t_sne.py

```diff
@@ -1132,6 +1150,28 @@ def fit_transform(self, X, y=None):
         X_new : ndarray of shape (n_samples, n_components)
             Embedding of the training data in low-dimensional space.
         """
+        # TODO(1.7): remove
+        # Also make sure to change `max_iter` default back to 1 and deprecate None
+        if self.n_iter != "deprecated":
+            if self.max_iter is not None:
+                raise ValueError(
+                    "Both 'n_iter' and 'max_iter' attributes were set. Attribute"
+                    " 'n_iter' was deprecated in version 1.5 and will be removed in"
+                    " 1.7. To avoid this error, only set the 'max_iter' attribute."
+                )
+            warnings.warn(
+                (
+                    "'n_iter' was renamed to 'max_iter' in version 1.5 and "
+                    "will be removed in 1.7."
+                ),
+                FutureWarning,
+            )
+            self.max_iter_ = self.n_iter
+        elif self.max_iter is None:
+            self.max_iter_ = 1000
+        else:
+            self.max_iter_ = self.max_iter
```

Comment:
It should be a private attribute ``self._max_iter``. I don't think we want to expose and document it.

Reply:
Sorry, yes you're right.